app/questionnaire/questionnaire_store_updater.py

```diff
@@ -35,8 +47,25 @@ def _update_questionnaire_store_with_form_data(self, form_data):
             if answer_id in answer_ids_for_question:
                 if answer_value not in self.EMPTY_ANSWER_VALUES:
                     answer = Answer(answer_id=answer_id,
+                                    list_item_id=self._current_location.list_item_id,
                                     value=answer_value)
 
                     self._answer_store.add_or_update(answer)
                 else:
-                    self._answer_store.remove(answer_ids=[answer_id])
+                    self._answer_store.remove_answer(answer_id)
+
+    def remove_all_answers_with_list_item_id(self, list_name: str, list_item_id: str):
+        """ Remove answers from the answer store and update the list store to remove it
+        """
+        self._list_store.delete_list_item_id(list_name, list_item_id)
+
+        self._answer_store.remove_all_answers_for_list_item_id(list_item_id=list_item_id)
+
+        self._questionnaire_store.add_or_update()
+        self._answer_store.dirty = True
```

Comment:
can we not set `dirty` in `_answer_store.remove_all_answers_for_list_item_id` as that knows if anything was removed?